app/scripts/lib/transaction/smart-transactions-mocks.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Mock for waitForTransactionHash. Simply replace the waitForTransactionHash
+ * with this mock so that we can debug locally without spending gas on mainnet.
+ *
+ * @returns Promise<string>
+ */
+export const mockWaitForTransactionHash: () => Promise<string> = () => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      // Need a real tx hash to pass some downstream validation
+      resolve(
+        '0xe3e223b9725765a7de557effdb2b507ace3534bcff2c1fe3a857e0791e56a518',
+      );
+    }, 20_000_000);
+  });
+};
+
+/**
+ * Mock for signAndSubmitTransactions. Simply replace the signAndSubmitTransactions
+ * with this mock so that we can debug locally without spending gas on mainnet.
+ *
+ * @returns Promise<{ uuid: string; txHash?: string }>
+ */
+export const mockSignAndSubmitTransactions: () => Promise<{
+  uuid: string;
+  txHash?: string;
+}> = () => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve({
+        uuid: 'uuid123456789',
+      });
+    }, 100);
```

Comment:
These functions are not supposed to be used in unit / integration tests, right?

Reply:
No it was just for testing out the flow without paying gas. I left them there for convenience sake, but we can take them out if you would like.

Reply:
If we would use them regularly when working on this feature I suppose we could keep them, but if they would be rarely used I would prefer to delete them to keep the codebase cleaner.